pkg/daemon/ceph/client/osd.go

```diff
@@ -355,3 +355,15 @@ func OSDOkToStop(context *clusterd.Context, clusterInfo *ClusterInfo, osdID, max
 
 	return stats.OSDs, nil
 }
+
+// OsdPrimaryAffinity assigns primary-affinity (within range [0.0, 1.0]) to a specific OSD.
+func OsdPrimaryAffinity(context *clusterd.Context, clusterInfo *ClusterInfo, osdID int, affinity string) error {
+	logger.Infof("setting osd.%d with primary-affinity %q", osdID, affinity)
+	args := []string{"osd", "primary-affinity", fmt.Sprintf("osd.%d", osdID), affinity}
+	_, err := NewCephCommand(context, clusterInfo, args).Run()
```

Comment:
Is it possible to set the primary affinity value as a command line arg to `ceph-osd` instead of by setting this property as a separate process?

Reply:
No. This is a command actually goes to the mon(s) before propagating to the osd.